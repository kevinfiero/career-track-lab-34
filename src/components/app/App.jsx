import React, { useReducer, useState } from 'react';
import reducer, { initialState } from '../../reducers/colorReducer';

function App() {
  const [state, dispatch] = useReducer(reducer, initialState)

  const handleChange = ({ target }) => {
    dispatch({
      type: target.id,
      payload: target.value
    });
  };

  const handleClick = ({ target }) => {
    dispatch({
      type: target.id
    });
  };

  return (
    <>
      <button id="UNDO" onClick={handleClick}>undo</button>
      <button id="REDO" onClick={handleClick}>redo</button>
      <input 
        id="RECORD"
        type="color" 
        alt="Color Picker" 
        value={state.current} 
        onChange={handleChange} 
      />
      <div 
        data-testid="Color Result" 
        style=
          {{ 
            backgroundColor: state.current, 
            width: '10rem', 
            height: '10rem' 
          }}>  
      </div>
    </>
  );
}

export default App;
