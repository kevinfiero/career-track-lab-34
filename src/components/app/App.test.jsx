import React from 'react';
import { render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('test for color selection', async() => {
    render(<App />);

    const colorInput = await screen.findByAltText('Color Picker');

    fireEvent.change(colorInput, {
      target: {
        value: '#00FF00'
      }
    });

    const display = await screen.findByTestId('Color Result');
    console.log(display.style.backgroundColor);
    
    expect(display).toHaveStyle({
      backgroundColor: '#00FF00'

    });
  });

  it('test for undo selection', async() => {
    render(<App />);

    const undoInput = await screen.findByText('undo');
    const colorInput = await screen.findByAltText('Color Picker');

    fireEvent.change(colorInput, {
      target: {
        value: '#00FF00'
      }
    });

    fireEvent.click(undoInput);

    const display = await screen.findByTestId('Color Result');
    console.log(display.style.backgroundColor);

    expect(display).toHaveStyle({
      backgroundColor: '#FF0000'

    });
  });

  it('test for redo selection', async() => {
    render(<App />);

    const undoInput = await screen.findByText('undo');
    const redoInput = await screen.findByText('redo');
    const colorInput = await screen.findByAltText('Color Picker');

    fireEvent.change(colorInput, {
      target: {
        value: '#00FF00'
      }
    });

    fireEvent.click(undoInput);
    fireEvent.click(redoInput);

    const display = await screen.findByTestId('Color Result');

    console.log(display.style.backgroundColor);

    expect(display).toHaveStyle({
      backgroundColor: '#00FF00'
    });
  });





});
