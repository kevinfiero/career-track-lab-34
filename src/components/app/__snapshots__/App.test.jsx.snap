// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <button>
    undo
  </button>
  <button>
    redo
  </button>
  <input
    alt="Color Picker"
    type="color"
    value="#FF0000"
  />
  <div
    data-testid="Color Result"
    style="background-color: rgb(255, 0, 0); width: 10rem; height: 10rem;"
  />
</DocumentFragment>
`;
